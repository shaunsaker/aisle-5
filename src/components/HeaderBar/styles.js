import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const HEADER_HEIGHT = 60;

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    paddingTop: styleConstants.helpers.isIphoneX ? 24 : styleConstants.dimensions.spacing.vertical,
    paddingBottom: styleConstants.dimensions.spacing.vertical,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal,
    backgroundColor: styleConstants.colors.primary,
    ...styleConstants.shadows.large,
    height: HEADER_HEIGHT,
    justifyContent: 'center',
  },
  networkActivityIndicatorContainer: {
    position: 'absolute',
    top: 0,
    right: styleConstants.dimensions.spacing.horizontal,
    bottom: 0,
    justifyContent: 'center',
  },
});

export default styles;
