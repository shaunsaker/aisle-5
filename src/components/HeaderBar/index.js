import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';

import styles from './styles';

import NetworkActivityIndicator from '../NetworkActivityIndicator';

const propTypes = {
  children: PropTypes.node,
  style: ViewPropTypes.style,
};

const defaultProps = {};

const HeaderBar = ({ children, style }) => {
  return (
    <View style={[styles.container, style]}>
      {children}
      <View style={styles.networkActivityIndicatorContainer}>
        <NetworkActivityIndicator />
      </View>
    </View>
  );
};

HeaderBar.propTypes = propTypes;
HeaderBar.defaultProps = defaultProps;

export default HeaderBar;
