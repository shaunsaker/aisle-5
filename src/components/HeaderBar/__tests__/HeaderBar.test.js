import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import HeaderBar from '..';

jest.mock('../../NetworkActivityIndicator', () => 'NetworkActivityIndicator');

describe('HeaderBar', () => {
  const children = <View />;
  const style = { backgroundColor: 'red' };

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(<HeaderBar style={style}>{children}</HeaderBar>);

      expect(component).toMatchSnapshot();
    });
  });
});
