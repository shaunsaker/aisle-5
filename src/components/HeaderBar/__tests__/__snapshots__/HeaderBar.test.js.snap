// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar renders renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#08293A",
        "borderColor": null,
        "borderWidth": 0,
        "elevation": 12,
        "height": 60,
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingHorizontal": 15,
        "paddingTop": 10,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 4,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View />
  <View
    style={
      Object {
        "bottom": 0,
        "justifyContent": "center",
        "position": "absolute",
        "right": 15,
        "top": 0,
      }
    }
  >
    <NetworkActivityIndicator />
  </View>
</View>
`;
